source/iterator/combined.go

```diff
@@ -270,7 +274,13 @@ func (c *Combined) getKeyColumn(ctx context.Context) (string, error) {
 // registerCustomVitessDialer registers a custom dialer. If the username and password arguments are provided,
 // GRPC authentication will be enabled.
 func registerCustomVitessDialer(ctx context.Context, username, password string) {
-	var grpcDialOptions []grpc.DialOption
+	var grpcDialOptions = []grpc.DialOption{
```

Comment:
Since the length of the `grpcDialOptions` is never 0, we don't need that check below anymore:

```go
if len(grpcDialOptions) == 0 {
	// if grpcDialOptions is nil we don't need to register a custom protocol,
	// we'll just use the default one.
	vitessProtocolName = *vtgateconn.VtgateProtocol

	return
}
```